watcher_test.go

```diff
@@ -604,6 +626,55 @@ func TestWatcherWait(t *testing.T) {
 	})
 }
 
+func TestWatcherMarkSweep(t *testing.T) {
+	t.Run("remove-old-dependency", func(t *testing.T) {
+		w := newWatcher(t)
+		defer w.Stop()
+		fdep := &idep.FakeDep{Name: "foo"}
+		bdep := &idep.FakeDep{Name: "bar"}
+		n := fakeNotifier("zed")
+		w.Register(n, fdep)
+		w.Register(n, bdep)
+
+		//
```

Comment:
Can you clean up these or add a comment?

Reply:
Meant to be the func/doc comment. I'll fill that in.

Reply:
🙌